pps_mw_training/pipelines/cloud_base/training_data.py

```diff
@@ -0,0 +1,142 @@
+    with xr.open_mfdataset(
+        [f.decode("utf-8") for f in data_files],
+        combine="nested",
+        concat_dim="nscene",
+    ) as all_data:
+        input_params = json.loads(input_parameters)
+        label_params = json.loads(label_parameters)
+        input_scaler = get_scaler(input_params)
+        input_data = np.stack(
+            [
+                input_scaler.apply(all_data[p["name"]].values, idx)
+                for idx, p in enumerate(input_params)
+            ],
+            axis=3,
+        )
+        label_scaler = get_scaler(label_params)
+        label_data = np.stack(
+            [
+                label_scaler.apply(all_data[p["name"]].values, idx)
+                for idx, p in enumerate(label_params)
+            ],
+            axis=3,
+        )
+        input_data[~np.isfinite(input_data)] = fill_value_input
+        label_data[~np.isfinite(label_data)] = fill_value_label
+        return [
+            np.float32(input_data),
+            np.float32(label_data),
+        ]
+
+
+@tf.function(
+    input_signature=(
+        tf.TensorSpec(shape=(None,), dtype=tf.string),
+        tf.TensorSpec(shape=(), dtype=tf.string),
+        tf.TensorSpec(shape=(), dtype=tf.string),
+        tf.TensorSpec(shape=(), dtype=tf.float32),
+        tf.TensorSpec(shape=(), dtype=tf.float32),
+    )
+)
+def load_data(
+    matched_files,
+    input_params,
+    label_params,
+    fill_value_input,
+    fill_value_label,
+):
+    """Load netcdf dataset."""
+    return tf.numpy_function(
+        func=_load_data,
+        inp=[
+            matched_files,
+            input_params,
+            label_params,
+            fill_value_input,
+            fill_value_label,
+        ],
+        Tout=[tf.float32, tf.float32],
+    )
+
+
+def _get_training_dataset(
+    matched_files: list[Path],
+    batch_size: int,
+    input_params: str,
+    label_params: str,
+    fill_value_input: float,
+    fill_value_label: float,
+) -> tf.data.Dataset:
+    """Get training dataset."""
+
+    ds = tf.data.Dataset.from_tensor_slices(
+        [f.as_posix() for f in matched_files]
+    )
+    ds = ds.batch(batch_size)
+    ds = ds.map(
+        lambda x: load_data(
+            x,
+            tf.constant(input_params),
+            tf.constant(label_params),
+            tf.constant(fill_value_input),
+            tf.constant(fill_value_label),
+        ),
+        num_parallel_calls=1,
+    )
+    return ds
+
+
+def get_training_dataset(
+    training_data_path: Path,
+    train_fraction: float,
+    validation_fraction: float,
+    test_fraction: float,
+    batch_size: int,
+    input_parameters: list[dict[str, Any]],
+    label_parameters: list[dict[str, Any]],
+    fill_value_input: float,
+    fill_value_label: float,
+) -> list[tf.data.Dataset]:
+    """Get training dataset."""
+
+    assert train_fraction + validation_fraction + test_fraction == 1
+    input_files = list((training_data_path).glob("cnn_data*.nc*"))[:1000]
```

Comment:
1000 should not be hard coded. better to remove it then. If we want to allow for changing it we should set it through the command line interface.

Reply:
I set it through the CLI but didnt like the way I did. I had to introduce another parameter in settings,py as FILE_LIMIT. Can there be a way to avoid this? Do you have a better suggestion?

Reply:
My suggestion is to remove the FILE_LIMIT from `settings.py` and let the defualt value be `None / all_files`. This will be the most clean and understandable logic.